### ISSUE TITLE
[Bug]: TypeError: Cannot read properties of undefined (reading 'map') in AutoHeight

### ISSUE THREAD

**Issue by suraj776** [davidjerleke/embla-carousel#1009]
### Which variants of Embla Carousel are you using?

- [ ] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [X] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Steps to reproduce

 I'm trying to create a integration test using React Testing library.
 
 Error:- I'm getting.
 
 **_detail: TypeError: Cannot read properties of undefined (reading 'map')
          at map (/node_modules/embla-carousel-auto-height/src/components/AutoHeight.ts:51:8)
          at highestInView (/node_modules/embla-carousel-auto-height/src/components/AutoHeight.ts:56:48)_**

      
    I tried all these mocks in my file. it didn't work. Also tried mocking the autoheight, then emblaCrousel started giving error.
    
    https://github.com/davidjerleke/embla-carousel/tree/master/packages/embla-carousel/src/__tests__/mocks

### Expected Behavior

It should be mockable when applying mocks.

### Additional Context

I have also tried...

### What browsers are you seeing the problem on?

All browsers

### Version

^8.2.0

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues
- [X] I agree to follow this project's Contributing Guidelines for bug reports

**Comment by davidjerleke** [comment]
@suraj776 have you created the necessary mocks? See:

- https://github.com/davidjerleke/embla-carousel/issues/837#issuecomment-2073969586
- #775

**Comment by suraj776** [comment]
@davidjerleke yes, i do have, all the mocks. Still getting error and it is from AutoHeight plugin.

**Comment by davidjerleke** [comment]
Thanks for clarifying @suraj776. I guess we need a null check [here](https://github.com/davidjerleke/embla-carousel/blob/master/packages/embla-carousel-auto-height/src/components/AutoHeight.ts#L48) in the codde.

**Comment by suraj776** [comment]
@davidjerleke yes it is working.

**Comment by davidjerleke** [comment]
@suraj776 no [here](https://github.com/davidjerleke/embla-carousel/blob/master/packages/embla-carousel-auto-height/src/components/AutoHeight.ts#L48), because we're mapping the `selectedIndexes` variable which is undefined. But that might lead to errors in other places. Like the function that depends on the `highestInView()` function:

```tsx
function setContainerHeight(): void {
  emblaApi.containerNode().style.height = `${highestInView()}px`
}
```

Can you create a CodeSandbox that reproduces the problem?

**Comment by suraj776** [comment]
@davidjerleke 

Please check this sandbox code.
https://codesandbox.io/p/devbox/elegant-moser-forked-psvcn3

### run command
`npm run test`

**Comment by davidjerleke** [comment]
@suraj776 thanks. That helps. So this solves it:

```tsx
function highestInView(): number | null {
  const { slideRegistry } = emblaApi.internalEngine()
  const selectedIndexes = slideRegistry[emblaApi.selectedScrollSnap()]

  if (!selectedIndexes) return null // Bail here if no selectedIndexes are found

  return selectedIndexes
    .map((index) => slideHeights[index])
    .reduce((a, b) => Math.max(a, b), 0)
}

function setContainerHeight(): void {
  const height = highestInView()
  if (height === null) return // Bail here if we don't have any height to apply

  emblaApi.containerNode().style.height = `${highestInView()}px`
}
```

Feel free to create a PR.

Best,
David

**Comment by davidjerleke** [comment]
@suraj776 I just released a bug fix for this in [v8.3.1](https://github.com/davidjerleke/embla-carousel/releases/tag/v8.3.1).
